### DISCUSSION TITLE
Distribution?

### DISCUSSION THREAD

**Discussion by afaulconbridge** [awslabs/linter-rules-for-nextflow#31] [General]
I want to run these via CI on a repository of Nextflow scripts.

But as far as I can tell, there is no distribution of built artefacts suitable for use in that way? I'll have to make that myself?

**Comment by markjschreiber** [comment]
What form of distribution would you need? Would you be able to use a container image similar to this one?https://github.com/awslabs/linter-rules-for-nextflow/blob/main/Dockerfile

**Comment by afaulconbridge** [comment]
For my use case, a GItHub Action like the ones in https://github.com/aws-actions would be ideal.

A close second is a container image built from the Dockerfile that already exists, in a public repository. Then I can use it where I need to in my own Actions.

A pre-built .jar that can be downloaded (similar to the ones for CodeNarc and SLF4J implementation mentioned in the readme) would also be helpful, to use in existing images such as devcontainers.

**Reply by markjschreiber** [reply]
There is a public container at https://gallery.ecr.aws/aws-genomics/linter-rules-for-nextflow

A jar can be built from the source with `./gradlew jar` but it won't contain the other dependencies, hence this section of the Dockerfile:

```
RUN wget https://repo1.maven.org/maven2/org/codenarc/CodeNarc/3.3.0/CodeNarc-3.3.0-all.jar && \
    wget https://repo1.maven.org/maven2/org/slf4j/slf4j-api/1.7.36/slf4j-api-1.7.36.jar && \
    wget https://repo1.maven.org/maven2/org/slf4j/slf4j-simple/1.7.36/slf4j-simple-1.7.36.jar
```
I can add the jar (without dependencies) to https://github.com/awslabs/linter-rules-for-nextflow/releases/tag/v0.1.0 if that helps.

**Comment by markjschreiber** [comment]
The linter rules jar is now included in https://github.com/awslabs/linter-rules-for-nextflow/releases/tag/v0.1.0 as is the ast-echo although you probably don't need that one.

**Comment by afaulconbridge** [comment]
Great, many thanks!

Perhaps you could add a link to https://gallery.ecr.aws/aws-genomics/linter-rules-for-nextflow in the readme? I checked Docker Hub and couldn't see anything, but wasn't familiar enough with https://gallery.ecr.aws to think of looking there.

**Reply by markjschreiber** [reply]
Good idea!
